compiler/rustc_metadata/src/creader.rs

```diff
@@ -1142,6 +1164,31 @@ fn format_dlopen_err(e: &(dyn std::error::Error + 'static)) -> String {
     e.sources().map(|e| format!(": {e}")).collect()
 }
 
+fn attempt_load_dylib(path: &Path) -> Result<libloading::Library, libloading::Error> {
+    #[cfg(target_os = "aix")]
+    if let Some(ext) = path.extension()
+        && ext.eq("a")
+    {
+        // On AIX, we ship all libraries as .a big_af archive
+        // the expected format is lib<name>.a(libname.so) for the actual
+        // dynamic library
+        let library_name = path.file_stem().expect("expect a library name");
+        let mut archive_member = OsString::from("a(");
+        archive_member.push(library_name);
+        archive_member.push(".so)");
+        let new_path = path.with_extension(archive_member);
+
+        // On AIX, we need RTLD_MEMBER to dlopen an archived shared
+        // flags = RTLD_LAZY | RTLD_LOCAL | RTLD_MEMBER
+        // these are not yet in libc https://github.com/rust-lang/libc/pull/4044
+        let flags = 0x4 | 0x80000 | 0x40000;
+        return unsafe { libloading::os::unix::Library::open(Some(&new_path), flags) }
+            .map(|lib| lib.into());
```

Comment:
Question: is this still accurate? the libc PR was merged but I dunno if we bumped the libc version in r-l/r yet.

Reply:
Thanks for pointing that out, it has been released a few days ago. Updating libc version